entities/entity_class.gd

```diff
@@ -51,6 +54,27 @@ func do_collect(entity: Entity):
 		status_sight_timer.start()
 		status_sight = true
 		ss_node.show()
+	if entity is Chest:
+		gold_pocket += 1
+		print(gold_pocket)
+		var ss_node = find_child("StatusGold")
+		if !ss_node:
+			return
+		if !status_gold_timer and gold_pocket >= 2:
+			print(gold_pocket, "Timer")
+			status_gold_timer = Timer.new()
+			status_gold_timer.wait_time = 10
+			status_gold_timer.one_shot = true
+			status_gold_timer.timeout.connect(func():
+				ss_node.hide()
+				status_gold = false
+			)
+			add_child(status_gold_timer)
+			status_gold_timer.start()
+			status_gold = true
+			ss_node.show()
```

Comment:
This part should be inside a separate `if` block
```gdscript
status_gold_timer.start()
status_gold = true
ss_node.show()
```
e.g. we need one `if` block to setup the timer (that only happens one time, if the timer doesn't exist alread), and another `if` block to check if `gold_pocket >= 2`